src/app/pages/questions/components/question/text-input/text-input.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
+
+@Component({
+  selector: 'text-input',
+  templateUrl: 'text-input.component.html'
+})
+export class TextInputComponent implements OnInit {
+  @Output()
+  valueChange: EventEmitter<string> = new EventEmitter<string>()
+
+  @Input()
+  type
+
+  placeholder = 'Enter text'
+  isDateTime: string
+  dateFormat: string
+
+  ngOnInit() {
+    this.isDateTime = this.getIsDateTime()
+    if (this.isDateTime) this.dateFormat = this.getDateFormat()
+  }
+
+  getIsDateTime() {
+    return this.type.includes('date') || this.type.includes('time')
+  }
+
+  getDateFormat() {
+    const type = this.type.split('_')[0]
+    const format = this.type.split('_')[1]
```

Comment:
Not sure about JS, but does this give an error if there is not format? (like `ArrayIndexOutOfBounds` in Java) or does it perhaps return undefined?
Personally, I prefer to move the initialisation of `format` inside the `datetime` switch clause as it is not used in any other clauses.

Reply:
Yup made the changes.